src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { NavLink } from "react-router-dom";
+import { Link, LinkGroup } from "./types";
+import Icon from "components/Icon";
+import { createNavLinkStyler } from "helpers";
+import Header from "./Header";
+
+type Props = {
+  endowId: number;
+  className?: string;
+  linkGroups: LinkGroup[];
```

Comment:
@misicnenad  
> Exact routes depend on the endowment ID that is connected

routes doesn't need to know the `id`, look at `adminRoutes`   
no `id` here. 
```
//no id here
"index"
/edit_profile
/settings
/invest
```

> The ID needs to be embedded somehow into the routes, so creating route constants would be unfeasible, you'd still end up with a creator function of some kind.

`react-router-dom ` `matchPath` takes care of route matching so no need to provide full path


> You need more than plain string routes to appropriately display the Sidebar items

I agree, I'm not saying we render icons and titles with just `string`, of course it's an object of some type
```ts
type RouteData = {
   icon: string
   size: number
   title: string
}
```
the question is do we pass this `RouteData` to `Sidebar` or just a minimum info like `/somePath` - which can be used to determine `RouteData` by itself where we have route data object where it can get this data from 
```ts
const routes = {
 "/edit_profile" : RouteData
 "/invest" : RouteData
 "/settings" : RouteData
}
```
having this `routes` constant common to `<Routes/>` and `<Layout/>` would mean we just need to past `keys` 
and there's common data
e.g
```tsx
//used in Route
<Route path={routes.edit_profile.url} />

//used in Nav
const route = routes[key]
<Nav  icon title /> 

```